### PR TITLE
Add territories to redis when ebook is unavailable

### PR DESCRIPTION
**Pull request by spaccagnini-vivlio**
Add a test when the ebook is not available to get the territories associated to this ebook

**Size**: S
**Jira**: https://tea-ebook.atlassian.net/browse/ONCALL-811

## Checklist

- [ ] I have verified that the commits are all meaningful.
- [ ] I have verified that changes are properly covered by automated tests according to the repository standard.
- [ ] I have verified that no documentation update or addition is needed or proposed such changes.

## Functional description

Add a test when the ebook is not available to get the territories associated to this ebook

### CODE REVIEW

File: lib/onix/product.rb [hunk 1]
```diff
@@ -555,6 +555,12 @@ def supplies(keep_all_prices_dates=false)
...
             end
+          elsif ps.markets && !ps.available?
+            if ps.markets.map{|m| m.territory.countries}.flatten.uniq.empty?
+                market_territories = []
+            else
+                market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
+            end
           else
             market_territories = []
           end
```
Comment (psdesse-vivlio) [inline lib/onix/product.rb]: I am having a hard time understanding these conditions. It looks like:
```
if ps.markets.map{|m| m.territory.countries}.flatten.uniq.empty?
    market_territories = []
else
    market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
end
```
is equivalent to:
```
market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
```

Would it be easier to read as:
```
if ps.markets
    if ps.markets.map{|m| m.territory.countries}.flatten.uniq.empty? && ps.available?
        market_territories = ONIX::Territory.region_to_countries("WORLD")
    else
        market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
    end
else
    market_territories = []
end
```

Ie. we have a specific condition when the product is available but as no market defined. Also we should put a long comment to explain the reason for this special case.

Comment (spaccagnini-vivlio) [reply lib/onix/product.rb]: yes, you're right, and I have simplified the code a little more

File: lib/onix/product.rb [hunk 2]
```diff
@@ -549,11 +549,10 @@ def supplies(keep_all_prices_dates=false)
...
         @product_supplies.each do |ps|
-          if ps.markets && ps.available?
-            if ps.markets.map{|m| m.territory.countries}.flatten.uniq.empty?
-                market_territories = ONIX::Territory.region_to_countries("WORLD")
-            else
-                market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
+          if ps.markets
+            market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
+            if market_territories.empty? && ps.available?
+              market_territories = ONIX::Territory.region_to_countries("WORLD")
```
Comment (psdesse-vivlio) [inline lib/onix/product.rb:557]: Yes great. Do you have from the last commit, the reason for this special case. I think we need a comment here to explain why.

**Review by psdesse-vivlio — APPROVED** [review]

File: lib/onix/product.rb (hunk 2 above)
Comment (spaccagnini-vivlio) [reply lib/onix/product.rb:557]: I add a comment

**Review by psdesse-vivlio — APPROVED** [review]

**Pull request merged**